ddtrace/internal/datastreams/processor.py

```diff
@@ -0,0 +1,270 @@
+            aggr_key = (",".join(edge_tags), hash_value, parent_hash)
+            stats = self._buckets[bucket_time_ns][aggr_key]
+            stats.full_pathway_latency.add(full_pathway_latency_sec)
+            stats.edge_latency.add(edge_latency_sec)
+
+    def _serialize_buckets(self):
+        # type: () -> List[Dict]
+        """Serialize and update the buckets."""
+        serialized_buckets = []
+        serialized_bucket_keys = []
+        for bucket_time_ns, bucket in self._buckets.items():
+            bucket_aggr_stats = []
+            serialized_bucket_keys.append(bucket_time_ns)
+
+            for aggr_key, stat_aggr in bucket.items():
+                edge_tags, hash_value, parent_hash = aggr_key
+                serialized_bucket = {
+                    u"EdgeTags": [six.ensure_text(tag) for tag in edge_tags.split(",")],
+                    u"Hash": hash_value,
+                    u"ParentHash": parent_hash,
+                    u"PathwayLatency": DDSketchProto.to_proto(stat_aggr.full_pathway_latency).SerializeToString(),
+                    u"EdgeLatency": DDSketchProto.to_proto(stat_aggr.edge_latency).SerializeToString(),
+                }
+                bucket_aggr_stats.append(serialized_bucket)
+            serialized_buckets.append(
+                {
+                    u"Start": bucket_time_ns,
+                    u"Duration": self._bucket_size_ns,
+                    u"Stats": bucket_aggr_stats,
+                }
+            )
+
+        # Clear out buckets that have been serialized
+        for key in serialized_bucket_keys:
+            del self._buckets[key]
+
+        return serialized_buckets
+
+    def _flush_stats(self, payload):
+        # type: (bytes) -> None
+        try:
+            conn = get_connection(self._agent_url, self._timeout)
+            conn.request("POST", self._endpoint, payload, self._headers)
+            resp = get_connection_response(conn)
+        except Exception:
+            log.error("failed to submit pathway stats to the Datadog agent at %s", self._agent_endpoint, exc_info=True)
+            raise
+        else:
+            if resp.status == 404:
+                log.error("Datadog agent does not support data streams monitoring. Upgrade to 7.34+")
+                self._enabled = False
+                return
+            elif resp.status >= 400:
+                log.error(
+                    "failed to send data stream stats payload, %s (%s) (%s) response from Datadog agent at %s",
+                    resp.status,
+                    resp.reason,
+                    resp.read(),
+                    self._agent_endpoint,
+                )
+            else:
+                log.debug("sent %s to %s", _human_size(len(payload)), self._agent_endpoint)
+
+    def periodic(self):
+        # type: () -> None
+
+        with self._lock:
+            serialized_stats = self._serialize_buckets()
+
+        if not serialized_stats:
```

Comment:
I wonder if we should debug log if there's nothing to send and this is enabled? From a solutions pov it'd probably be helpful to see that there's nothing to send, rather than there not being any log line at all, which would bring up the question of if this is enabled/running properly. If it's only every 10 seconds that we flush if there's nothing, I don't think it should be an issue from a performance standpoint.  Thoughts?